components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/user/OpenIDConnectUserEndpoint.java

```diff
@@ -70,6 +71,10 @@ public Response getUserClaims(@Context HttpServletRequest request) throws OAuthS
                     UserInfoEndpointConfig.getInstance().getUserInfoAccessTokenValidator();
             OAuth2TokenValidationResponseDTO tokenResponse = tokenValidator.validateToken(accessToken);
 
+            if (tokenResponse.isValid() && !isValidTokenBinding(tokenResponse.getTokenBinding(), request)) {
```

Comment:
do we need to have tokenResponse.isValid() ?  It might not be logical to do this check and then throw an error saying "Valid token binding value not present in the request."

Reply:
fixed.